### PR TITLE
feat: Added PowerShell checker pre-commit hook

### PR DESCRIPTION
**Pull request by jakub-kocka**
- Added pre-commit PowerShell hook such as Python wrapper for PowerShell script to be able to run on all platforms
- When PowerShell is not present in the system the error message to install it is displayed - this will happen only if the hook runs (any .ps1 script will be edited)

Tested on Windows and Linux (WSL)

This hook  can be tested locally in different repositories for example ESP-IDF with:
`pre-commit try-repo <path_to_repo> check-powershell-scripts`

However, the files have to be already uploaded to this repository (so I have tested with a different repository).

OUTPUT:

    pre-commit try-repo ..\personal_GH_esp_tests\ check-powershell-scripts
    [WARNING] Creating temporary repo with uncommitted changes...
     ==========================
    Using config:
    ==========================
    repos:
    -   repo: C:\Users\JAKUBK~1\AppData\Local\Temp\tmp5yxji34v\shadow- 
    repo
        rev: e3ab2a638d221adb0dc4530822dcb4272f080623
        hooks:
        -   id: check-powershell-scripts
    ==========================
    [INFO] Initializing environment for C:\Users\JAKUBK~1\AppData\Local\Temp\tmp5yxji34v\shadow-repo.
    [INFO] Installing environment for C:\Users\JAKUBK~1\AppData\Local\Temp\tmp5yxji34v\shadow-repo.
    [INFO] Once installed this environment will be reused.
    [INFO] This may take a few minutes...
    Check PowerShell Scripts.................................................Failed
    - hook id: check-powershell-scripts
    - exit code: 1
    
    Windows PowerShell
    Copyright (C) Microsoft Corporation. All rights reserved.
    
    Install the latest PowerShell for new features and improvements! https://aka.ms/PSWindows
    
    
    PS C:\Projects\esp-idf> Traceback (most recent call last):
      File "<frozen runpy>", line 198, in _run_module_as_main
      File "<frozen runpy>", line 88, in _run_code
      File "C:\Users\JAKUBK~1\AppData\Local\Temp\tmp5yxji34v\reposwxlzq_h\py_env-python3.11\Lib\site-packages\esp_pwsh_check\__main__.py", line 33, in <module>
    main()
      File "C:\Users\JAKUBK~1\AppData\Local\Temp\tmp5yxji34v\reposwxlzq_h\py_env-python3.11\Lib\site-packages\esp_pwsh_check\__main__.py", line 29, in main
    raise SystemError("[ERROR] PowerShell Script error")
    SystemError: [ERROR] PowerShell Script error
     
    Installing PSScriptAnalyzer
    Checking: export.ps1
    PowerShell Script Error
    
    RuleName                            Severity     ScriptName Line  Message
    --------                            --------     ---------- ----  -------
    PSAvoidAssignmentToAutomaticVariabl Error        export.ps1 32    Starting from PowerShell 6.0, the Variable
    e                                                                 'IsWindows' cannot be assigned any more since it is a
                                                                  readonly automatic variable that is built into
                                                                  PowerShell, please use a different name.
    PSPossibleIncorrectComparisonWithNu Warning      export.ps1 29    $null should be on the left side of equality
    ll                                                                comparisons.
    PSPossibleIncorrectComparisonWithNu Warning      export.ps1 63    $null should be on the left side of equality
    ll                                                                comparisons.

### CODE REVIEW

**Comment by jakub-kocka** [comment]
Can you please take a look, @mfialaf?

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+  ```sh
+  virtualenv venv && source ./venv/bin/activate
+  ```
+
+- Install the project and development dependencies:
+  ```sh
+  pip install -e '.[dev]'
+  ```
+
```
Comment (mfialaf) [inline CONTRIBUTING.md]: If I wanted to contribute, I would appreciate an example command on how to execute the check after this local setup. Something as is in this PR description.

**Review by mfialaf — APPROVED** [review]
Good job Jakub. LGTM!

File: CONTRIBUTING.md (hunk 1 above)
Comment (jakub-kocka) [reply CONTRIBUTING.md]: I will update this, thanks for your review.

**Comment by jakub-kocka** [comment]
Can you please take a look @tomassebestik?

**Review by tomassebestik — APPROVED** [review]
Wrapper and the project structure is ok.

I'm not entirely sure about the necessity of having PowerShell installed on the system for this to function. It may be the easiest option here, but doing it like this expect lot of hassle.

I've noticed some pre-commit hooks use an isolated environment in Docker (I think  `hadolint` pre-commit hook doing this). 

Docker is commonly available on most developers' machines (PowerShell is not on most of Linux and MAC machines). Running this within a pre-commit CI job, which is already containerized, might be complex and quite tricky.

Letting the decision you to you.

**Comment by jakub-kocka** [comment]
Thank you for your reviews. I have decided to use the Docker approach and containerize the application @tomassebestik.
The function should be unchanged, I have tested this on Windows and on Ubuntu in WSL.

**Comment by jakub-kocka** [comment]
I have updated the `README.md` and `CONTRIBUTING.md` according to the usage of Docker implementation.

I am merging this based on an in-person discussion.

**Pull request merged**